lxd/db/certificates.go

```diff
@@ -27,23 +27,28 @@ import (
 //go:generate mapper stmt -p db -e certificate update struct=Certificate
 //
 //go:generate mapper method -p db -e certificate List
-//go:generate mapper method -p db -e certificate Get
+//go:generate mapper method -p db -e certificate List-by-Fingerprint
```

Comment:
@masnax The `certificates` table has a `UNIQUE (fingerprint)` on fingerprint, so we don't need a `List-by-Fingerprint`, instead the existing `List` function should accept a filter with a `Fingerprint` field.

Reply:
I think part of the problem here is that for certificates, the "fingerprint" can be just a fingerprint prefix, and so it *can* return multiple rows potentially. Although technically speaking that isn't the "fingerprint" but the "fingerprint prefix".

My thoughts is that for 'special case' searching like that (by prefix) it would be easier to have a single hand-written function for finding the 'real' certificate fingerprint by prefix, and then having the rest of the generated functions operate only on the real fingerprint.

Reply:
> @masnax The `certificates` table has a `UNIQUE (fingerprint)` on fingerprint, so we don't need a `List-by-Fingerprint`, instead the existing `List` function should accept a filter with a `Fingerprint` field.

Currently I'm blocking fields from the struct if they are used in a `-by-` statement, and `certificates` also has a `DeleteOne-by-Fingerprint` field, so if I were to add `fingerprint` to the filter struct, the delete statement would be `DeleteOne`, which would accept any field from the filter struct. If we're OK with that, I'll make the change.